### ISSUE TITLE
Add an Android-like screenshot button to the workspaces view

### ISSUE THREAD

**Issue by hanaral** [elementary/gala#954]
## Problem
Sometimes people feel that they want a UI way of screenshotting, that wouldn't require them to open a seperate app.

## Proposal
Add a screenshot button to the workspaces overview, possibly as a small round bubble in a corner of the worspace preview (like the notification dissmiss one)

## Prior Art
Maybe something like this, but with a more suitable and visible icon
![scrin](https://user-images.githubusercontent.com/58219504/97818031-ec001580-1c97-11eb-9ab0-9856a8d08003.png)

**Comment by fabiozaramella** [comment]
I actually like the idea but I also think that any normal icon would end up cluttering the UI. Maybe it would look good with a symbolic icon inside the monitor preview's rectangle in the same corner as shown above and showing it only when hovering the preview with the mouse (or not for touch support? 🤷🏽‍♂️). 

Leaving my consideration here for @elementary/ux team. 😉

**Comment by janxkoci** [comment]
Wouldn't this take a screenshot of windows spread instead of normal desktop view?

And what's wrong with pressing `PrintScreen` key?

**Comment by hanaral** [comment]
> Wouldn't this take a screenshot of windows spread instead of normal desktop view?
The window spread is drawn by Gala, and I presume that it could screenshot them without the spread in the background 
> And what's wrong with pressing `PrintScreen` key?
>Sometimes people feel that they want a UI way of screenshotting
>that wouldn't require them to open a seperate app

**Comment by janxkoci** [comment]
> Sometimes people feel that they want a UI way of screenshotting that wouldn't require them to open a seperate app.

Fair enough. Although I still have to say that taking screenshots on linux in general (and elementary OS in particular) is waaaaaaay more easy and intuitive than other platforms, which use weird unmemorable shortcuts (macOS) or weird workflows (Windows, which usually requires pasting the screenshot into some app for saving - most people use Word, which ruins the image by compression).

Anyway, I'm not sure if the workspace view is the best place for this feature. For start, the screenshot will look differently than what user actually sees, if it's implemented as suggested:

> The window spread is drawn by Gala, and I presume that it could screenshot them without the spread in the background

That can be confusing or just make it hard to arrange windows properly. Moreover, what if somebody want's to make screenshot of the workspace overview _itself_? You add extra options? That would lead to overly complicated and cumbersome solution in my view.

I think simpler approach would be to just **show screenshot tool icon on the dock by default**. The app doesn't really feel like app to me anyway. Showing it on dock makes it more notable for new users while old-timers can easily remove it from dock, use shortcut or whatever they are comfortable with.

**Comment by hanaral** [comment]
@janxkoci You're still quite right about the workspace screenshot implying that it would screenshot the workspace view and not the desktop. 

> I think simpler approach would be to just show screenshot tool icon on the dock by default. The app doesn't really feel like app to me anyway. Showing it on dock makes it more notable for new users while old-timers can easily remove it from dock, use shortcut or whatever they are comfortable with.

I'm not sure how to go about it, since it wouldn't be a sin to put the app in the dock, but it also feels like something that has an option to be done from some form of system function (like the overview).
Maybe the dock really is the way to go?

**Comment by janxkoci** [comment]
Since you mentioned android, I thought maybe it could be put in some indicator (e.g. the user indicator comes to mind), but it still felt out of place in the current indicators and making a new one just for this feels like an overkill.

Then I realized that **slingshot** is just another indicator, in a way, and you can launch the screenshot tool from there. So there is that.

And also note that the **shortcut overview** after pressing the Super key (my key has a little Tux penguin there) also mentions shortcuts for taking screenshots.

I really think that taking screenshots on elementary OS is plenty easy already :smiley: Adding the dock icon could be useful for beginners; another improvement would be a notification showing where the picture is saved (and open it upon clicking). Another alternative could be an icon as a **plugin for the plank dock** (there are 5 already shipping with plank now) which would directly take a screenshot of the desktop, instead of launching the tool. This will probably be less straightforward as plank is packaged upstream and not developed by elementary.

**Comment by hanaral** [comment]
I just thought of something that seems like good idea; how about adding a window screenshot button to one of the corners of a window when you hover over it in the workspace view? That way it's consistent with the close-window badge that behaves the same way, and also makes what it'll do obvious
